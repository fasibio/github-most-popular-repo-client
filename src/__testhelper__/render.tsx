import { ReactElement } from 'react';
import {
  render as realRender,
  fireEvent,
  RenderOptions,
  RenderResult,
} from '@testing-library/react';

import { Grommet } from 'grommet';

export const render = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
): RenderResult => {
  return realRender(<Grommet>{ui}</Grommet>, options);
};
