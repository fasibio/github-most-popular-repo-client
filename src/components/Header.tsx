import { Box, Button, Header, Menu, ResponsiveContext, Text } from 'grommet';
import { Moon, Paint, Sun } from 'grommet-icons';
import React from 'react';

import { useThemeing, useThemeMode } from '../store/themeing';

export const AppHeader = () => {
  return (
    <Header
      height={'50px'}
      width={'100%'}
      background="brand"
      justify={'end'}
      flex={false}
    >
      <ThemeingNav />
    </Header>
  );
};

export const ThemeingNav = () => {
  const { themes, setSelectedTheme, selectedTheme } = useThemeing();
  const size = React.useContext(ResponsiveContext);
  const { dark, setMode } = useThemeMode();
  return (
    <Box direction="row" height={'50px'} justify="end">
      <Button
        hoverIndicator
        onClick={() => setMode(!dark)}
        icon={dark ? <Sun /> : <Moon />}
      />
      <Box height="50px" justify="center" align="center" pad="small">
        <Menu
          dropBackground={'brand'}
          dropProps={{ align: { top: 'top', right: 'right' } }}
          items={themes.map((theme) => {
            return {
              label: (
                <Box
                  align="center"
                  gap="small"
                  margin={'small'}
                  justify="center"
                  alignSelf="center"
                >
                  {theme.name}
                </Box>
              ),
              onClick: () => {
                setSelectedTheme(theme);
              },
              active: selectedTheme.name === theme.name,
            };
          })}
        >
          <Box
            align="end"
            gap={'small'}
            margin={{ right: 'small' }}
            justify="center"
            direction="row"
          >
            <Paint />
            {size !== 'small' && <Text>{selectedTheme.name}</Text>}
          </Box>
        </Menu>
      </Box>
    </Box>
  );
};
