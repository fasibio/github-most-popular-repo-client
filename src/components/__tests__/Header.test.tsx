import { AppHeader } from '../Header';
import { render } from '../../__testhelper__/render';

describe('tests header component', () => {
  it('component render snapshot', () => {
    const component = render(<AppHeader />);
    expect(component).toMatchSnapshot();
  });
});
