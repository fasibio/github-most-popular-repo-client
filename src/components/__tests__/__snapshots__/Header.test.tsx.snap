// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests header component component render snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="StyledGrommet-sc-19lkkz7-0 ioQEen"
      >
        <header
          class="StyledBox-sc-13pk1d4-0 bvXgxI"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 gQjVgV"
          >
            <button
              class="StyledButton-sc-323bzc-0 ijXFTb"
              type="button"
            >
              <svg
                aria-label="Sun"
                class="StyledIcon-sc-ofa7kd-0 wnZwe"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 4V2m0 20v-2m8-8h2M2 12h2m13.657-5.657L19.07 4.93M4.93 19.07l1.414-1.414m0-11.314L4.93 4.93m14.14 14.14-1.414-1.414M12 17a5 5 0 1 0 0-10 5 5 0 0 0 0 10z"
                  stroke="#000"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
            <div
              class="StyledBox-sc-13pk1d4-0 iPTQfK"
            >
              <button
                aria-expanded="false"
                aria-haspopup="menu"
                aria-label="Open Menu"
                class="StyledButton-sc-323bzc-0 hIPuDF"
                type="button"
              >
                <div
                  class="StyledBox-sc-13pk1d4-0 jMudjv"
                >
                  <svg
                    aria-label="Paint"
                    class="StyledIcon-sc-ofa7kd-0 wnZwe"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M4 10H2c0-5.523 0-9 10-9s10 3.477 10 9h-2m-8 0c4.418 0 8-.895 8-2s-3.582-2-8-2-8 .895-8 2 3.582 2 8 2zM4 20c0 1.657 3.582 3 8 3s8-1.343 8-3m0-12v12V8zM4 20V8v12zm4-7v4m5-4v4m-6-5c-1.5 0-3-1-3-4m14 5.5V12c0-2 2-1 2-4M8 13a1 1 0 0 0-1-1m6 1a1 1 0 0 1 2 0v.5m0 0a1.5 1.5 0 0 0 3 0M8 17a2.5 2.5 0 1 0 5 0"
                      fill="none"
                      stroke="#000"
                      stroke-width="2"
                    />
                  </svg>
                </div>
              </button>
            </div>
          </div>
        </header>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="StyledGrommet-sc-19lkkz7-0 ioQEen"
    >
      <header
        class="StyledBox-sc-13pk1d4-0 bvXgxI"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 gQjVgV"
        >
          <button
            class="StyledButton-sc-323bzc-0 ijXFTb"
            type="button"
          >
            <svg
              aria-label="Sun"
              class="StyledIcon-sc-ofa7kd-0 wnZwe"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 4V2m0 20v-2m8-8h2M2 12h2m13.657-5.657L19.07 4.93M4.93 19.07l1.414-1.414m0-11.314L4.93 4.93m14.14 14.14-1.414-1.414M12 17a5 5 0 1 0 0-10 5 5 0 0 0 0 10z"
                stroke="#000"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
          <div
            class="StyledBox-sc-13pk1d4-0 iPTQfK"
          >
            <button
              aria-expanded="false"
              aria-haspopup="menu"
              aria-label="Open Menu"
              class="StyledButton-sc-323bzc-0 hIPuDF"
              type="button"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 jMudjv"
              >
                <svg
                  aria-label="Paint"
                  class="StyledIcon-sc-ofa7kd-0 wnZwe"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M4 10H2c0-5.523 0-9 10-9s10 3.477 10 9h-2m-8 0c4.418 0 8-.895 8-2s-3.582-2-8-2-8 .895-8 2 3.582 2 8 2zM4 20c0 1.657 3.582 3 8 3s8-1.343 8-3m0-12v12V8zM4 20V8v12zm4-7v4m5-4v4m-6-5c-1.5 0-3-1-3-4m14 5.5V12c0-2 2-1 2-4M8 13a1 1 0 0 0-1-1m6 1a1 1 0 0 1 2 0v.5m0 0a1.5 1.5 0 0 0 3 0M8 17a2.5 2.5 0 1 0 5 0"
                    fill="none"
                    stroke="#000"
                    stroke-width="2"
                  />
                </svg>
              </div>
            </button>
          </div>
        </div>
      </header>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
